### ISSUE TITLE
Define tool to look for dead / unused tokens to remvoe

### ISSUE THREAD

**Issue by pylapp** [Orange-OpenSource/ouds-ios#1237]
### Discussed in https://github.com/Orange-OpenSource/ouds-ios/discussions/1180

<div type='discussions-op-text'>

<sup>Originally posted by **pylapp** November 26, 2025</sup>
# Context

Today with for example version v0.21.0 we have **1,971 tokens (403 core raw tokens, 79 Orange brand raw tokens, 50 Sosh brand raw tokens, 139 Wireframe brand raw tokens, 967 core semantic tokens, 333 core component tokens)**.
These tokens are mostly defined in Figma, then exposed thanks to the _tokenator_ in Swift code.

But it is possible some of these tokens are not used, maybe because an issue in the conception of the component or because in the end the token is not mandatory.

Reducing the number of tokens will reduced the weight of the repository to be cloned when projects downloads it. In addition, it will reduce the memory used to store them for nothing.

Removing useless tokens will help users find the ones they need without being polutted / disturbed by the number of tokens too big.

Dead code tools like _periphery_ are not enough as because tokens are public it will ignore them.

Related to https://github.com/Orange-OpenSource/ouds-ios/discussions/935

# Definition of Done

- [ ] Make the Python script
- [ ] Write the list of tokens (here for example) not used
- [ ] Check with design where they should be used
- [ ] If not used, remove them from tokenator and code bases
- [ ] Compute the memory saved and the weight of the repo
</div>

### Steps

For the Python script:
- define a Python 3.8 script in _scripts_ folder, with the suitable shebang and sources header
- declare an array of files names (e.g. supposed to protocols declaring tokens) to process
- define an array of patterns to look for and use to extract only the token name
- define an array of file extensions (e.g. containing ".swift") at least to use as filters to process only files with these extensions
- store in an object the tokens: stored objects must contain the token name, the protocol file names (to know where it comes from)
- then, for each token, and for all files, check if the token appears but beware: the line containing the supposed available token must not be a comment (e.g. not starting by // or ///)
- if a token is missing, i.e. not available at list in one Swift source code file, add this token and the protocol file name in another object
- in the end, in logs, display the tokens not used in the Swift code base
- using a map defining bytes sizes for Swift types, compute the naive amount of memory which can be saved if these tokens are removed
- if no token is missing, return system exit 0. If a token is missing return system exist 1. If any error occurs return system exit 2.
- source code must be documented in english with Python standards
-  Make a Python script, in _scripts_ folder of the library, extracting all component, semantic and raw tokens (from protocols and enums extensions). Then, for each of these token, check if the token is used (i.e. excluded tests and tokens providers). If not, log it.
